crates/amaru-stores/src/rocksdb/columns/dreps.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::rocksdb::common::{as_key, as_value, PREFIX_LEN};
+use amaru_ledger::store::StoreError;
+use rocksdb::Transaction;
+use tracing::error;
+
+use amaru_ledger::store::columns::dreps::{Key, Row, Value, EVENT_TARGET};
+
+/// Name prefixed used for storing DReps entries. UTF-8 encoding for "drep"
+pub const PREFIX: [u8; PREFIX_LEN] = [0x64, 0x72, 0x65, 0x70];
+
+/// Register a new DRep.
+pub fn add<DB>(
+    db: &Transaction<'_, DB>,
+    rows: impl Iterator<Item = (Key, Value)>,
+) -> Result<(), StoreError> {
+    for (credential, (anchor, deposit)) in rows {
+        let key = as_key(&PREFIX, &credential);
+
+        // In case where a registration already exists, then we must only update the underlying
+        // entry.
+        if let Some(mut row) = db
+            .get(&key)
+            .map_err(|err| StoreError::Internal(err.into()))?
+            .map(Row::unsafe_decode)
+        {
+            row.anchor = anchor;
+            if let Some(deposit) = deposit {
+                row.deposit = deposit;
+            }
+            db.put(key, as_value(row))
+                .map_err(|err| StoreError::Internal(err.into()))?;
+        }  else if let Some(deposit) = deposit {
+            let row = Row {
+                anchor,
+                deposit,
+            };
+            db.put(key, as_value(row))
+                .map_err(|err| StoreError::Internal(err.into()))?;
+        } else {
+            error!(
```

Comment:
@KtorZ Replicating the behavior from `accounts` now it appears to me that both always have a deposit? This logic derives from `DiffBind` storing values as `Option`.

Reply:
So technically:

- DReps always have an explicit deposits
- Accounts *may* have an explicit deposits; historically, they had not. 

Although if I recall correctly how this was implemented in Amaru, we are _defaulting_ to the protocol parameters' deposit value when the deposit is unspecified. So from a storage perspective, they should indeed always have a deposit. 

That's a simplification we could do on the DiffBind structure; I am honestly not quite sure why it's still an option. Maybe trying to remove it will re-surface the reason why ^^"